File: programmers/js/level3/멀리_뛰기.js
function solution(n) {
    let dp = Array.from(Array(n));
    dp[0] = 1;
    if (n > 1) {
        dp[1] = 2;
    }
    for (let i = 2; i < n; ++i) {
        dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;
    }
    return dp[n - 1];
}

console.log(solution(4));

File: programmers/old/level3/멀리_뛰기.java
class Solution {
	public long solution(int n) {
		if (n == 1) {
			return 1;
		}
		long[] arr = new long[n];
		arr[0] = 1;
		arr[1] = 2;
		for (int i = 2; i < n; ++i) {
			arr[i] = (arr[i - 1] + arr[i - 2]) % 1234567;
		}
		return arr[n - 1];
	}
}

public class 멀리_뛰기 {
	public static void main(String[] args) {
		int n = 4;
		Solution sol = new Solution();
		System.out.println(sol.solution(n));
	}
}
